### PR TITLE
Use PNPM

### PR DESCRIPTION
**Pull request by KhudaDad414**
**Description**
- Following #1060 this PR migrates studio to PNPM.
- Some other quality of life improvements.

* [x] Use PNPM in test workflow
* [x] Use PNPM in the deploy workflow
* [x] Remove `nodejs` tag from the repo after merge

### CODE REVIEW

**Comment by KhudaDad414** [comment]
@Amzani @Shurtu-gal. PR is ready to review.

> PS: Sorry for the long PR. It was like opening a can of worms. 😆

**Comment by Shurtu-gal** [comment]
@KhudaDad414 there are some conflicts

**Comment by KhudaDad414** [comment]
@Shurtu-gal the conflicts has been resolved.

**Review by Shurtu-gal — APPROVED** [review]

File: .github/workflows/bounty-program-commands.yml
[the remark this answers is not in the document]
Comment (KhudaDad414) [reply .github/workflows/bounty-program-commands.yml:1]: unfortunately we can't. `.github` only supports `npm`.

if everything goes well, we can purpose our changes to `.github` repo and make sure other packages in AsyncAPI can also use `pnpm` as their package manager.

I have also remove the `nodejs` tag which makes sure that we do not get updates for the test script.

**Review by Amzani — CHANGES_REQUESTED** [review]
Looks good, only small change regarding this sentence in the README

```
Note: NPM v7+ is required.
```

Change to pnpm

File: .github/workflows/bounty-program-commands.yml
[the remark this answers is not in the document]
Comment (Shurtu-gal) [reply .github/workflows/bounty-program-commands.yml:1]: That's fine I was only speaking about this specific file but as it is the same with the `.github` one there is no issue. I guess I was seeing commit diff leading to this file being shown as modified.

**Review by Shurtu-gal — APPROVED** [review]

**Comment by KhudaDad414** [comment]
@Amzani done.

**Review by Amzani — APPROVED** [review]
LGTM

**Pull request merged**